src/Web/Handler.hs

```diff
@@ -147,6 +200,161 @@ getTheory idx = do
     liftIO $ withMVar (theoryVar yesod) $ return. M.lookup idx
 
 
+getLemmaPlaintext :: Int -> TheoryPath -> Handler String
+getLemmaPlaintext nr path = do
+    let lname = case path of 
+            (TheoryEdit n) -> n 
+            _ -> "this should never be visible"
+    eitherTheory <- getTheory nr
+    let lemmaItem = case eitherTheory of 
+            (Just (Trace thy)) -> lookupLemma lname (tiTheory thy)
+            _ -> Nothing
+    let plaintext =  fromMaybe "Enter your new Lemma" $ get lPlaintext <$> lemmaItem
+    return plaintext
+
+
+getStartingProof ::  System -> IncrementalProof
+getStartingProof gsys = LNode (ProofStep (Sorry Nothing) (Just gsys)) M.empty
+
+
+editProof :: Int -> String -> Handler (Either String TheoryIdx)
+editProof idx name = do
+    let result = withTheory idx $ \ti -> do
```

Comment:
So here you define `result` in a let-construct in a do-block, but you do-block only has this let command and then returns that variable. This is redundant. I'd rather define the `\ti -> do ...` block as a an explicit function in `where` and then simply define `editProof` as `withTheory idx THAT_NEW_FUNCTION`.